runtime/src/consensus/tendermint/verifier.rs

```diff
@@ -248,11 +251,19 @@ impl Verifier {
         instance: &mut Instance,
     ) -> Result<ConsensusState, Error> {
         let height = cache.latest_known_height();
+        self.consensus_state_at(instance, height)
+    }
+
+    fn consensus_state_at(
+        &self,
+        instance: &mut Instance,
+        height: u64,
+    ) -> Result<ConsensusState, Error> {
         let verified_block = instance
             .light_client
             .verify_to_target(height.try_into().unwrap(), &mut instance.state)
             .map_err(|err| Error::VerificationFailed(err.into()))?;
-        // No need to update time as this has already been done once for this height.
+        self.update_insecure_posix_time(&verified_block);
```

Comment:
Should we update the cache also? 
`cache.update_verified_block(&verified_block);`

Seems like `verify_to_target`, `update_insecure_posix_time` and `update_verified_block` should always be called together, maybe in the future we refactor and group them.

Reply:
Both good ideas, updated. And we can do the suggested refactor in a subsequent PR.